universe/auto_syncer.go

```diff
@@ -313,6 +452,51 @@ func (f *FederationEnvoy) syncer() {
 				continue
 			}
 
+			// After we've synced with the federation, we'll
+			// attempt to push out any pending proofs that we
+			// haven't yet completed.
+			ctxFetchLog, cancelFetchLog := f.WithCtxQuitNoTimeout()
+			syncDirection := SyncDirectionPush
+			logEntries, err :=
+				f.cfg.FederationDB.FetchPendingProofsSyncLog(
+					ctxFetchLog, &syncDirection,
+				)
+			if err != nil {
+				log.Warnf("unable to query pending push "+
+					"sync log: %w", err)
+				continue
+			}
+			cancelFetchLog()
```

Comment:
We should also cancel right before the `continue` above. Or just move this cancel to directly after the DB call.

Reply:
Good point.